packages/peregrine/package.json

```diff
@@ -47,5 +47,8 @@
   "module": "lib/index.js",
   "jsnext:main": "lib/index.js",
   "es2015": "lib/index.js",
-  "sideEffects": false
+  "sideEffects": false,
+  "dependencies": {
+    "tapable": "~1.1.3"
```

Comment:
Is it necessary for us to import a dependency for what is effectively just an event/callback model? If so, do we want to use a dependency that seems to be written for server-side operations? We would need to ensure that the code as written would work in our supported browsers.

I'm also not sure how much this adds to the bundle but it would be worth looking into.

Reply:
Your points are valid. It might have been written keeping the server in mind. But this is far from being a simple event/callback model. This supports many more things that I have not demonstrated in this PR. Things like injecting, hijacking, piggybacking, bailing function internals in both sync and async modes. To do something like this on our own would take lot of time and resources.

Regarding browser compatibility, I don't have a lot of information to share on this topic.

Reply:
The intended use case of this package is to allow 3rd party extensions to have ability to control function implementations and return values in Venia UI and Peregrine.

Reply:
>This supports many more things that I have not demonstrated in this PR. Things like injecting, hijacking, piggybacking, bailing function internals in both sync and async modes.

I would love to see examples of this.